airflow/api_fastapi/core_api/routes/public/variables.py

```diff
@@ -166,6 +167,21 @@ def post_variable(
     session: SessionDep,
 ) -> VariableResponse:
     """Create a variable."""
+    # Check for key length limit
+    if len(post_body.key) > ID_LEN:
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=f"Variable key exceeds maximum length of {ID_LEN} characters.",
+        )
```

Comment:
This shouldn't be done manually like that. Pydantic have native validators for that.
Something like that in the datamodel should do the exact same.

```
Field(max_length=256)
```

Reply:
I kept it at 250, since this was given on the API before, also used the same const to keep single source of truth.